src/resources/subscription.ts

```diff
@@ -0,0 +1,68 @@
+import axios from 'axios'
+import { SafepayConfig, SubscriptionCreateData } from '../types'
+import { buildApiUrl } from '../utils/builder'
+
+export class Subscription {
+  private config: SafepayConfig
+
+  constructor(config: SafepayConfig) {
+    this.config = config
+  }
+
+  async cancel(subscriptionId: string): Promise<Subscription> {
+    const url = buildApiUrl(this.config.environment)
+    const {
+      data: { data }
+    } = await axios.request<{
+      data: Subscription
+    }>({
+      baseURL: url,
+      headers: { 'X-SFPY-MERCHANT-SECRET': this.config.v1Secret },
+      method: 'post',
+      url: `/subscriptions/v1/${subscriptionId}/cancel`,
+      data: {}
+    })
+
+    return data
+  }
+
+  async pause({
+    subscriptionId,
+    behavior
+  }: {
+    subscriptionId: string
+    behavior: string
```

Comment:
shouldn't this be one of the enums you have declared in the other file? how are you validating that the `behavior` param being passed in is valid